lfs/certs.go

```diff
@@ -0,0 +1,135 @@
+package lfs
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/github/git-lfs/vendor/_nuts/github.com/rubyist/tracerx"
+)
+
+// getRootCAsForHost returns a certificate pool for that specific host (which may
+// be "host:port" loaded from either the gitconfig or from a platform-specific
+// source which is not included by default in the golang certificate search)
+// May return nil if it doesn't have anything to add, in which case the default
+// RootCAs will be used if passed to TLSClientConfig.RootCAs
+func getRootCAsForHost(host string) *x509.CertPool {
+
+	// don't init pool, want to return nil not empty if none found; init only on successful add cert
+	var pool *x509.CertPool
+
+	// gitconfig first
+	pool = appendRootCAsForHostFromGitconfig(pool, host)
+	// Platform specific
+	return appendRootCAsForHostFromPlatform(pool, host)
+
+}
+
+func appendRootCAsForHostFromGitconfig(pool *x509.CertPool, host string) *x509.CertPool {
+	// Accumulate certs from all these locations:
+
+	loaded := false
+	// GIT_SSL_CAINFO first
+	if cafile, ok := Config.envVars["GIT_SSL_CAINFO"]; ok {
+		pool = appendCertsFromFile(pool, cafile)
+		loaded = true
+	}
```

Comment:
Is there reason this isn't just returning that `pool`? Seems like we could remove all these `loaded` checks.